### PR TITLE
Remove old version of JanDeDobbeleer.OhMyPosh

### LINKED ISSUE #26177: The validation in sandbox command should not run as admin by default
# Description of the new feature/enhancement
The validation script should not use admin console to run in sandbox.

Take `tencent.tencentmeeting` as an example. This package can only really install the content files when the installer is run from a console with admin privilage. If you run it from a user mode console, the installer will actually install nothing, even `--scope machine` added. However, this installer returns nothing even under situation. So winget will believe it installed successfully. But in sandbox, all the console will start as admin, which cannot reflect what we use console normally (normally we start a console as normal user). So we need to start the validation script as usermode and check whether new programs was added.
# Proposed technical implementation details (optional)
Let the validation script start a console in user mode in sandbox and validation pipeline.

### PR DESCRIPTION
**Pull request by vedantmgoyal9**
null

### CODE REVIEW

**Comment by vedantmgoyal2009** [comment]
cc @lychichem @OfficialEsco

**Comment by jedieaston** [comment]
```diff
diff --git a/Tools/SandboxTest.ps1 b/Tools/SandboxTest.ps1
index e78320953..62f22cd41 100644
--- a/Tools/SandboxTest.ps1
+++ b/Tools/SandboxTest.ps1
@@ -119,7 +119,7 @@ if (-Not [String]::IsNullOrWhiteSpace($Manifest)) {
 
 Write-Host '--> Checking dependencies'
 
-$desktopInSandbox = 'C:\Users\WDAGUtilityAccount\Desktop'
+$desktopInSandbox = 'C:\Public\'
 
 foreach ($dependency in $dependencies) {
   $dependency.file = Join-Path -Path $tempFolder -ChildPath $dependency.fileName
@@ -246,7 +246,7 @@ $switchingUserScriptContent | Out-File $switchingUserScriptFile
 $switchingUserScriptFileInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $switchingUserScriptFileName)
 
 $mapFolderInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Split-Path -Path $mapFolder -Leaf)
-
+$tempFolderInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Split-Path -Path $tempFolder -Leaf)
 # Create Wsb file
 $sandboxTestWsbContent = @"
 <Configuration>
@@ -254,9 +254,11 @@ $sandboxTestWsbContent = @"
     <MappedFolder>
       <HostFolder>$tempFolder</HostFolder>
       <ReadOnly>true</ReadOnly>
+      <SandboxFolder>$tempFolderInSandbox</SandboxFolder>
     </MappedFolder>
     <MappedFolder>
       <HostFolder>$mapFolder</HostFolder>
+      <SandboxFolder>$mapFolderInSandbox</SandboxFolder>
     </MappedFolder>
   </MappedFolders>
   <LogonCommand>

```

**Comment by jedieaston** [comment]
```diff
diff --git a/Tools/SandboxTest.ps1 b/Tools/SandboxTest.ps1
--- Tools/SandboxTest.ps1
+++ Tools/SandboxTest.ps1
@@ -118,13 +118,13 @@
 # Download dependencies
 
 Write-Host '--> Checking dependencies'
 
-$desktopInSandbox = 'C:\Users\WDAGUtilityAccount\Desktop'
+$sharedFolderInSandbox = 'C:\Shared\'
 
 foreach ($dependency in $dependencies) {
   $dependency.file = Join-Path -Path $tempFolder -ChildPath $dependency.fileName
-  $dependency.pathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $dependency.fileName)
+  $dependency.pathInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $dependency.fileName)
 
   # Only download if the file does not exist, or its hash does not match.
   if (-Not ((Test-Path -Path $dependency.file -PathType Leaf) -And $dependency.hash -eq $(get-filehash $dependency.file).Hash)) {
     Write-Host @"
@@ -182,9 +182,9 @@
 "@
 
 if (-Not [String]::IsNullOrWhiteSpace($Manifest)) {
   $manifestFileName = Split-Path $Manifest -Leaf
-  $manifestPathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $manifestFileName)
+  $manifestPathInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $manifestFileName)
 
   $bootstrapPs1Content += @"
 Write-Host @'
 
@@ -227,9 +227,9 @@
 
 $bootstrapPs1FileName = 'Bootstrap.ps1'
 $bootstrapPs1Content | Out-File (Join-Path -Path $tempFolder -ChildPath $bootstrapPs1FileName)
 
-$bootstrapPs1InSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $bootstrapPs1FileName)
+$bootstrapPs1InSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $bootstrapPs1FileName)
 
 # Switch to Local User
 $switchingUserScriptContent = @"
 `$localusrname = 'wingettest';
@@ -242,22 +242,24 @@
 "@
 $switchingUserScriptFileName = 'SwitchLocalUser.ps1'
 $switchingUserScriptFile = Join-Path -Path $tempFolder -ChildPath $switchingUserScriptFileName
 $switchingUserScriptContent | Out-File $switchingUserScriptFile
-$switchingUserScriptFileInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $switchingUserScriptFileName)
+$switchingUserScriptFileInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $switchingUserScriptFileName)
 
-$mapFolderInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Split-Path -Path $mapFolder -Leaf)
-
+$mapFolderInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Split-Path -Path $mapFolder -Leaf)
+$tempFolderInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Split-Path -Path $tempFolder -Leaf)
 # Create Wsb file
 $sandboxTestWsbContent = @"
 <Configuration>
   <MappedFolders>
     <MappedFolder>
       <HostFolder>$tempFolder</HostFolder>
       <ReadOnly>true</ReadOnly>
+      <SandboxFolder>$tempFolderInSandbox</SandboxFolder>
     </MappedFolder>
     <MappedFolder>
       <HostFolder>$mapFolder</HostFolder>
+      <SandboxFolder>$mapFolderInSandbox</SandboxFolder>
     </MappedFolder>
   </MappedFolders>
   <LogonCommand>
   <Command>PowerShell Start-Process PowerShell -WindowStyle Maximized -WorkingDirectory '$mapFolderInSandbox' -ArgumentList '-ExecutionPolicy Bypass -NoExit -NoLogo -File $switchingUserScriptFileInSandbox'</Command>

```

**Comment by jedieaston** [comment]
```diff
diff --git a/Tools/SandboxTest.ps1 b/Tools/SandboxTest.ps1
--- Tools/SandboxTest.ps1
+++ Tools/SandboxTest.ps1
@@ -118,13 +118,13 @@
 # Download dependencies
 
 Write-Host '--> Checking dependencies'
 
-$desktopInSandbox = 'C:\Users\WDAGUtilityAccount\Desktop'
+$sharedFolderInSandbox = 'C:\Shared\'
 
 foreach ($dependency in $dependencies) {
   $dependency.file = Join-Path -Path $tempFolder -ChildPath $dependency.fileName
-  $dependency.pathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $dependency.fileName)
+  $dependency.pathInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $dependency.fileName)
 
   # Only download if the file does not exist, or its hash does not match.
   if (-Not ((Test-Path -Path $dependency.file -PathType Leaf) -And $dependency.hash -eq $(get-filehash $dependency.file).Hash)) {
     Write-Host @"
@@ -182,9 +182,9 @@
 "@
 
 if (-Not [String]::IsNullOrWhiteSpace($Manifest)) {
   $manifestFileName = Split-Path $Manifest -Leaf
-  $manifestPathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $manifestFileName)
+  $manifestPathInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $manifestFileName)
 
   $bootstrapPs1Content += @"
 Write-Host @'
 
@@ -227,9 +227,9 @@
 
 $bootstrapPs1FileName = 'Bootstrap.ps1'
 $bootstrapPs1Content | Out-File (Join-Path -Path $tempFolder -ChildPath $bootstrapPs1FileName)
 
-$bootstrapPs1InSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $bootstrapPs1FileName)
+$bootstrapPs1InSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $bootstrapPs1FileName)
 
 # Switch to Local User
 $switchingUserScriptContent = @"
 `$localusrname = 'wingettest';
@@ -242,22 +242,24 @@
 "@
 $switchingUserScriptFileName = 'SwitchLocalUser.ps1'
 $switchingUserScriptFile = Join-Path -Path $tempFolder -ChildPath $switchingUserScriptFileName
 $switchingUserScriptContent | Out-File $switchingUserScriptFile
-$switchingUserScriptFileInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $switchingUserScriptFileName)
+$switchingUserScriptFileInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $switchingUserScriptFileName)
 
-$mapFolderInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Split-Path -Path $mapFolder -Leaf)
-
+$mapFolderInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Split-Path -Path $mapFolder -Leaf)
+$tempFolderInSandbox = Join-Path -Path $sharedFolderInSandbox -ChildPath (Split-Path -Path $tempFolder -Leaf)
 # Create Wsb file
 $sandboxTestWsbContent = @"
 <Configuration>
   <MappedFolders>
     <MappedFolder>
       <HostFolder>$tempFolder</HostFolder>
       <ReadOnly>true</ReadOnly>
+      <SandboxFolder>$tempFolderInSandbox</SandboxFolder>
     </MappedFolder>
     <MappedFolder>
       <HostFolder>$mapFolder</HostFolder>
+      <SandboxFolder>$mapFolderInSandbox</SandboxFolder>
     </MappedFolder>
   </MappedFolders>
   <LogonCommand>
   <Command>PowerShell Start-Process PowerShell -WindowStyle Maximized -WorkingDirectory '$mapFolderInSandbox' -ArgumentList '-ExecutionPolicy Bypass -NoExit -NoLogo -File $switchingUserScriptFileInSandbox'</Command>

```

**Comment by vedantmgoyal2009** [comment]
What are these comments? I could not understand at all.

**Review by jedieaston — CHANGES_REQUESTED** [review]
Some comments:

- The way you have it now, the user account you create can't actually see any of the shared folders because they are under `C:\Users\WDAGUtilityAccount`. I changed `$desktopInSandbox` to a folder they can both see and [changed the Sandbox config](https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-sandbox/windows-sandbox-configure-using-wsb-file#mapped-folders) to mount the folders there. It works as expected, the user account you make executes `winget install`.


- Even after you do that, since UAC is disabled system wide in the sandbox, the account you create can't ask for escalation when running an installer so anything not user-scoped can't run. Making this as it stands right now a bit pointless if it is designed to solve the issue (since @lychichem as I understood it needed a way to test user to admin escalation). 

Here's a rant that is largely irrelevant to your PR, but I've been trying to figure it out all afternoon and didn't come up with anything:

(I should start a discussion about it), but we need to find a better way to test manifests. The sandbox isn't designed for this and I don't think is cutting it anymore with all of these edge cases. Until the containers team can explain why they won't give us images with GUI support I don't have a better solution though (VMs take too long to boot). But having to reinstall winget every time I boot a sandbox, only being able to run one at a time (why? You can run multiple WDAG containers at once, and it's the same stuff underneath), only being able to run as admin unless you get around it like this, not being able to reboot. It's starting to become an inaccurate place to test compared to the validation environment, which is getting annoying. 

/rant

**Comment by jedieaston** [comment]
Applying the above diff to the script and running it with the latest Microsoft.EdgeBeta manifest gets me this, btw: 
image

(because msiexec can't ask for elevation since there's no UAC)

**Comment by OfficialEsco** [comment]
> (I should start a discussion about it), but we need to find a better way to test manifests. The sandbox isn't designed for this and I don't think is cutting it anymore with all of these edge cases. Until the containers team can explain why they won't give us images with GUI support I don't have a better solution though (VMs take too long to boot). But having to reinstall winget every time I boot a sandbox, only being able to run one at a time (why? You can run multiple WDAG containers at once, and it's the same stuff underneath), only being able to run as admin unless you get around it like this, not being able to reboot. It's starting to become an inaccurate place to test compared to the validation environment, which is getting annoying.

Create a Discussion, personally i use a VM and do NOT reset for each PR, if they install dependency's i uninstall those after testing the PR. I believe Levvie does the same according to how her script is setup [Community Moderators - 2021-08-27 - experimental.cmd](https://github.com/ItzLevvie/winget-pkgs-validate-and-install/blob/main/Community%20Moderators%20-%202021-08-27%20-%20experimental.cmd)
If i'm at work i use Sandbox, however i still import that script.

IMO [PRTest.ps1](https://github.com/microsoft/winget-pkgs/blob/master/Tools/PRTest.ps1) and [SandboxTest.ps1](https://github.com/microsoft/winget-pkgs/blob/master/Tools/SandboxTest.ps1) is only for simple/quick testing, not for Moderators to testing 10+ PRs

Somewhat related Discussion https://github.com/microsoft/winget-pkgs/discussions/1349

**Comment by vedantmgoyal2009** [comment]
Can't we convert #26177 to the discussion?

**Comment by vedantmgoyal2009** [comment]
I've started a discussion https://github.com/microsoft/winget-pkgs/discussions/26267

**Review by OfficialEsco — APPROVED** [review]
